src/Model/ModelManager.php

```diff
@@ -443,7 +454,7 @@ public function modelReverseTransform($class, array $array = [])
 
             if ($reflClass->hasMethod($setter)) {
                 if (!$reflClass->getMethod($setter)->isPublic()) {
-                    throw new PropertyAccessDeniedException(sprintf('Method "%s()" is not public in class "%s"', $setter, $reflClass->getName()));
+                    throw new \RuntimeException(sprintf('Method "%s()" is not public in class "%s"', $setter, $reflClass->getName()));
```

Comment:
IMO, `BadMethodCallException` could be a better choice here; but `DomainException` should work for both (the method and the property).